src/app/containers/VestedHistory/index.tsx

```diff
@@ -111,8 +121,21 @@ export function VestedHistory() {
             setEventsHistoryVestingOrigin(newRes);
           });
       }
+      if (vestingFourYear.value !== ethGenesisAddress) {
+        fouryear = await eventReader
+          .getPastEvents('staking', 'TokensStaked', {
+            staker: vestingFourYear.value,
+          })
+          .then(res => {
+            const newRes = res.map(v => ({
+              ...v,
+              type: t(translations.stake.currentVests.assetType.fouryear),
+            }));
+            setEventsHistoryVestingFourYear(newRes);
+          });
+      }
       try {
-        Promise.all([reward, genesis, team, origin]).then(_ =>
+        Promise.all([reward, genesis, team, origin, fouryear]).then(_ =>
           setLoading(false),
         );
```

Comment:
Use either `await` or `Promise.all`. The values in the `Promise.all([reward, genesis, team, origin, fouryear])` should be all undefined if I read this correctly and skips directly to the `then`

Reply:
You are totally correct, good catch - no idea why the original code was written like this